Exercises/.cmake/FindKokkos.cmake

```diff
@@ -0,0 +1,44 @@
+# Early return if Kokkos is already set up
+# We do not use Kokkos_FOUND as it is not globally defined
+if (TARGET Kokkos::kokkos)
+    return()
+endif ()
+
+set(SPACK_CXX $ENV{SPACK_CXX})
+if (SPACK_CXX)
+    message("found spack compiler ${SPACK_CXX}")
+    set(CMAKE_CXX_COMPILER ${SPACK_CXX} CACHE STRING "the C++ compiler" FORCE)
+    set(ENV{CXX} ${SPACK_CXX})
```

Comment:
I find it confusing when this environment variable is silently set. It seems it's sufficient to set `CMAKE_CXX_COMPILER` as a cache variable for the exercises.
```suggestion
```

Reply:
I want to remove this old spack stuff in a separate PR.